src/node_buffer.cc

```diff
@@ -606,6 +606,9 @@ void Fill(const FunctionCallbackInfo<Value>& args) {
 
   } else if (enc == UCS2) {
     node::TwoByteValue str(env->isolate(), args[1]);
+    if (IsBigEndian())
+      SwapBytes16(reinterpret_cast<char*>(&str[0]), str_length);
```

Comment:
Sorry, I should have made this clearer, but this is already the working version; i.e. this line was *missing* and should already have been there afaict